lib/utils/gestures.js

```diff
@@ -334,6 +331,15 @@ function untrackDocument(stateObj) {
 // Use passive event listeners, if supported, to not affect scrolling performance
 document.addEventListener('touchend', ignoreMouse, SUPPORTS_PASSIVE ? {passive: true} : false);
 
+/**
+ * Returns the composedPath for the given event.
+ * @param {Event} event to process
+ * @return {!Array<!EventTarget>} Path of the event
+ */
+const getComposedPath = window.ShadyDOM && window.ShadyDOM.noPatch ?
+  window.ShadyDOM.composedPath :
+  (event) => event.composedPath && event.composedPath() || [];
```

Comment:
Order of operations is confusing, is this always guaranteed to coerce to a `[]`?

Maybe change last expression to a ternary?

Reply:
Minor, disregard